basic_autonomy/src/basic_autonomy.cpp

```diff
@@ -234,6 +234,198 @@ namespace basic_autonomy
+            reference_centerline.insert(reference_centerline.end(), current_lanelet_centerline.begin(), current_lanelet_centerline.end());
+
+            while(!shared_boundary_found){
+
+                if(current_lanelet.leftBound() == ending_lanelet.rightBound()){   
+                    is_lanechange_left = true;
+                    shared_boundary_found = true;
+                }
+
+                else if(current_lanelet.rightBound() == ending_lanelet.leftBound()){
+                    shared_boundary_found = true;
+                }
+
+                else{
+                    //In case there are no following lanelets on route, lanechange should be completing before reaching it
+                    if(wm->getMapRoutingGraph()->following(current_lanelet, false).empty())
+                    {
+                        throw(std::invalid_argument("No following lanelets from current lanelet reachable without a lane change, incorrectly chosen end lanelet"));
+                    }
+
+                    current_lanelet = wm->getMapRoutingGraph()->following(current_lanelet, false).front(); 
+                    auto current_lanelet_linestring = current_lanelet.centerline2d().basicLineString();    
+                    reference_centerline.insert(reference_centerline.end(), current_lanelet_linestring.begin(), current_lanelet_linestring.end());
+                    starting_lane.push_back(current_lanelet);
+                }
+            }
+
+            //Find crosstrack distance between starting lane and ending lane
+            std::vector<lanelet::BasicPoint2d> ending_lane_centerline;
+            for(size_t i = 0;i<starting_lane.size();++i){
+                lanelet::ConstLanelet curr_end_lanelet;
+                
+                if(is_lanechange_left){
+                    
+                    //get left lanelet
+                    if(wm->getMapRoutingGraph()->left(starting_lane[0])){
+                        curr_end_lanelet = wm->getMapRoutingGraph()->left(starting_lane[0]).get();
+                    }
+                    else{
+                        curr_end_lanelet = wm->getMapRoutingGraph()->adjacentLeft(starting_lane[0]).get();
+                    }
+                }
+                else{
+
+                    //get right lanelet
+                    if(wm->getMapRoutingGraph()->right(starting_lane[0])){
+                        curr_end_lanelet = wm->getMapRoutingGraph()->right(starting_lane[0]).get();
+                    }
+                    else{
+                        curr_end_lanelet = wm->getMapRoutingGraph()->adjacentRight(starting_lane[0]).get();
+                    }
+                }
+                
+                auto ending_lane_linestring = curr_end_lanelet.centerline2d().basicLineString();
+                ending_lane_centerline.insert(ending_lane_centerline.end(), ending_lane_linestring.begin(), ending_lane_linestring.end());
+                
+            }
```

Comment:
Under the current logic, I think there is a case missing in this loop for when we are already in the ending lane. This plugin will still be getting called until the lane change is actually complete. However, if we did not rely on vehicle state and used the start, end lanelets from the maneuver as mentioned in my earlier comment this would not be an issue.

Reply:
Has this been addressed?

Reply:
Yes, we are using the starting downtrack for the maneuver to plan the lane change trajectory, like suggested in the second part of the comment